File: benchmarks/Java/src/Benchmark.java
public abstract class Benchmark {

    public abstract Object benchmark();
    public abstract boolean verifyResult(Object result);
    public abstract boolean setUp(int innerIterations);

    public boolean innerBenchmarkLoop(final int innerIterations) {
        for (int i = 0; i < innerIterations; i++) {
            if (!verifyResult(benchmark())) {
                return false;
            }
        }
        return true;
    }
}

File: benchmarks/scroll/src/main/scala/Benchmark.scala
import scala.util.boundary, boundary.break

abstract class Benchmark {

  def benchmark(): Object
  def verifyResult(result: Object): Boolean
  def setUp(innerIterations: Int): Boolean

  def innerBenchmarkLoop(innerIterations: Int): Boolean = {
    boundary:
      for (i <- 1 to innerIterations)
        if !verifyResult(benchmark()) then break()

    return true
  }

}
